code/API_definitions/Edge-Application-Management.yaml

```diff
@@ -961,6 +999,278 @@ components:
+      properties:
+        primaryNetwork:
+          description: Definition of Kubernetes primary Network
+          type: object
+          properties:
+            provider:
+              description: CNI provider name
+              type: string
+              example: cilium
+            version:
+              description: CNI provider version
+              type: string
+              example: "1.13"
+        additionalNetworks:
+          description: Additional Networks for the Kubernetes cluster.
+          type: array
+          items:
+            type: object
+            description: Additional network interface definition
+            properties:
+              name:
+                description: Additional Network Name
+                type: string
+                example: net1
+              interfaceType:
+                description: |
+                  Type of additional Interface:
+                  netdevice: (SR-IOV) A regular kernel network device in the
+                             Network Namespace (netns) of the container
+                  vfio-pci: (SR-IOV) A PCI network interface directly mounted
+                            in the container
+                  interface: Additional interface to be used by cni plugins
+                             such as macvlan, ipvlan
+                  Note: The use of SR-IOV interfaces automatically
+                  configure the required kernel parameters for the nodes.
+                type: string
+                example: vfio-pci
+                enum:
+                  - netdevice
+                  - vfio-pci
+                  - interface
+
+    AdditionalStorage:
+      description: Additional storage for the application.
+      type: array
+      items:
+        type: object
+        required:
+          - storageSize
+          - mountPoint
+        properties:
+          name:
+            type: string
+            description: Name of additional storage resource.
+            example: logs
+          storageSize:
+            type: string
+            description: Additional persistent volume for the application.
+            example: 80GB
+            pattern: ^\d+(GB|MB)$
+          mountPoint:
+            type: string
+            description: Location of additional storage resource.
+            example: /logs
+
+    Vcpu:
+      type: string
+      pattern: ^\d+((\.\d{1,3})|(m))?$
+      description: |
+        Number of vcpus in whole (i.e 1), decimal (i.e 0.500) up to
+        millivcpu, or millivcpu (i.e 500m) format.
+      example: "500m"
+
+    KubernetesResources:
+      description: Definition of Kubernetes Cluster Infrastructure.
+      type: object
+      required:
+        - nodePools
+        - infraKind
+        - applicationResources
+      properties:
+        infraKind:
+          description: Type of infrastructure for the application.
+          type: string
+          example: kubernetes
+          enum:
+            - kubernetes
```

Comment:
The infrakind is part of the top level attribute KubernetesResources and looks redundant with value as "kubernetes" as KubernetesResources itself indicate that it is kubernetes resource.